src/microbe_stage/Microbe.cs

```diff
@@ -904,10 +904,29 @@ public void AIThink(float delta, Random random, MicrobeAICommonData data)
     /// </returns>
     public List<(Compound Compound, Color Colour, Vector3 Target)> GetDetectedCompounds(CompoundCloudSystem clouds)
     {
+        HashSet<(Compound Compound, float Range, float MinAmount, Color Colour)> collectedUniqueCompoundDetections;
+
+        if (Colony != null)
```

Comment:
This needs to be so that this *only* triggers for the colony leader. I just playtested and this'll be the last needed change.

Reply:
If you can't make this change right away, I'll make the change so that I can finish testing this and get on with my other branch.